exonum-java-binding-core/src/main/java/com/exonum/binding/service/Service.java

```diff
@@ -118,4 +118,13 @@
    * @param router a router responsible for handling requests to this service
    */
   void createPublicApiHandlers(Node node, Router router);
+
+  /**
+   * Handles block commit event. This handler is a callback method which is invoked by
+   * the blockchain after each block commit.
+   * For example, a service can create one or more transactions if a specific condition
+   * has occurred.
+   * @param event the context allowing to access the blockchain state as of that committed block
+   */
+  default void afterCommit(BlockCommittedEvent event) {}
```

Comment:
As this method may be used for anything (it does not have a narrow purpose, as #convertTx or Tx#execute), I'd specify what will happen (or reassure that nothing bad will) if it throws some exception.

Reply:
Do you think we should specify it for this method? I've added `read-only` and `optional` to this javadoc, I think it should imply that it's okay to throw here. If that's not enough, could you please propose the correct wording?

Reply:
Mb something like that: 
`... it's safe to throw any exceptions inside the method and it won't cause transaction failing. `

Reply:
Doesn't seem correct to me - what transaction are we speaking about? Maybe we should refer to `block commit` rather than `transaction`?

Reply:
As it is invoked during block processing (afaik, in the same thread), I'd reassure that any exceptions will be swallowed and will not affect the processing of transactions/blocks.